haystack/templates/pipeline.py

```diff
@@ -115,41 +128,74 @@ def override(self, component_name: str, component_instance: Component) -> "Pipel
         `component_instance` is not a valid component.
         """
         # check if the component_name is allowed in the template
-        if component_name not in self.templated_variables:
+        if component_name not in self.template_variables:
             raise PipelineValidationError(f"Component '{component_name}' is not defined in the pipeline template")
+
         if not isinstance(component_instance, Component):
             raise PipelineValidationError(
-                f"'{type(component_instance)}' doesn't seem to be a component. Is this class decorated with @component?"
+                f"'{type(component_instance)}' is not a Component. Is this class decorated with @component?"
             )
-        self.components[component_name] = component_to_dict(component_instance)
+
+        self.component_overrides[component_name] = yaml.safe_dump(component_to_dict(component_instance))
```

Comment:
We inject into the templates the YAML representation of the overriding component